pie/trainer.py

```diff
@@ -150,19 +150,25 @@ def get_weights(self):
 
 
 class LRScheduler(object):
-    def __init__(self, optimizer, threshold=0.0, **kwargs):
-        self.lr_scheduler = optim.lr_scheduler.ReduceLROnPlateau(
-            optimizer, mode='max', threshold=threshold, **kwargs)
+    def __init__(self, optimizer, lr_scheduler='ReduceLROnPlateau', **kwargs)
+        self.lr_scheduler = getattr(optim.lr_scheduler, lr_scheduler)(
+            optimizer, **kwargs)
 
     def step(self, score):
-        self.lr_scheduler.step(score)
+        if isinstance(self.lr_scheduler, optim.lr_scheduler.ReduceLROnPlateau):
+            self.lr_scheduler.step(score)
+        else:
+            slef.lr_scheduler.step()
 
     def __repr__(self):
-        return '<LrScheduler lr="{:g}" steps="{}" patience="{}" threshold="{}"/>' \
-            .format(self.lr_scheduler.optimizer.param_groups[0]['lr'],
-                    self.lr_scheduler.num_bad_epochs,
-                    self.lr_scheduler.patience,
-                    self.lr_scheduler.threshold)
+        res = '<LrScheduler lr="{:g}"'.format(
+            self.lr_scheduler.optimizer.param_groups[0]['lr'])
+        for key in dir(self.lr_scheduler):
```

Comment:
We are losing some manual filter on what we want to show here

Reply:
Sure, but otherwise you have to do it manually per class...

Reply:
I know... Couldn't we curate for specific scheduler ? And automatically show for others ?

Reply:
as long as it doesn't introduce too much clutter, feel free!